5.4.connected_vessels.py

```diff
@@ -0,0 +1,37 @@
+# The function accepts one or more iterable parameters, and returns a list of the interlaced elements
+def interleave(*iterable):
+    if not iterable:
+        return None
+    res = []
+    j = 0
+
+    # If we haven't reached the end of the first element
+    # - assuming that the size of all elements is the same
+    while j < len(iterable[0]):
+        for i in iterable:
+            if type(i) is dict:  # Checking if the element is a dictionary
+                res += [list(i)[j]]
+
+            else:
+                res += [i[j]]
+        j += 1
+    return res
+
+
+# Version with generator
+def interleave_generator(*iterable):
+    j = 0
```

Comment:
Consider replacing 'j', 'i' and 'res' with more explanatory variable names, Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

Reply:
I fixed it.